mathext/hypergeo_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright ©2016 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"fmt"
+	"math"
+	"testing"
+)
+
+func TestHyp2f1(t *testing.T) {
+	t.Parallel()
+
+	// Constants taken from https://github.com/RobinHankin/hypergeo/blob/master/tests/testthat/test_aaa.R
+	var tests = []struct {
+		x float64
+		y float64
+	}{
+		{x: 0.28, y: 1.3531156987873853569937},
+		{x: -0.79, y: 0.5773356740314405932679},
+		{x: 0.56, y: 2.1085704049533617876477},
+		{x: -2.13, y: 0.3352446571148822718200},
+		{x: -0.43, y: 0.7150355048137748692483},
+		{x: -1.23, y: 0.4670987707934830535095},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			t.Parallel()
+			y := Hypergeo(1.21, 1.443, 1.88, test.x)
+			if d := y - test.y; math.Abs(d) > 1e-12 {
```

Comment:
```suggestion
			if !scalar.EqualWithinAbs(y, test.y, 1e-12) {
```

Reply:
Thanks for making me aware of `EqualWithinAbs`, which is a very useful and powerful tool.
Applied here and elsewhere.